lib/internal/url.js

```diff
@@ -255,33 +255,41 @@ class URL {
   }
 
   [util.inspect.custom](depth, opts) {
+    if (this == null ||
+        Object.getPrototypeOf(this[context]) !== URLContext.prototype) {
+      throw new TypeError('Value of `this` is not a URL');
+    }
+
     const ctx = this[context];
-    var ret = 'URL {\n';
-    ret += `  href: ${this.href}\n`;
-    if (ctx.scheme !== undefined)
-      ret += `  protocol: ${this.protocol}\n`;
-    if (ctx.username !== undefined)
-      ret += `  username: ${this.username}\n`;
-    if (ctx.password !== undefined) {
-      const pwd = opts.showHidden ? ctx.password : '--------';
-      ret += `  password: ${pwd}\n`;
-    }
-    if (ctx.host !== undefined)
-      ret += `  hostname: ${this.hostname}\n`;
-    if (ctx.port !== undefined)
-      ret += `  port: ${this.port}\n`;
-    if (ctx.path !== undefined)
-      ret += `  pathname: ${this.pathname}\n`;
-    if (ctx.query !== undefined)
-      ret += `  search: ${this.search}\n`;
-    if (ctx.fragment !== undefined)
-      ret += `  hash: ${this.hash}\n`;
+
+    if (typeof depth === 'number' && depth < 0)
+      return opts.stylize('[Object]', 'special');
+
+    const obj = Object.assign(new URLTemp(), {
+      href: this.href,
+      origin: this.origin,
+      protocol: this.protocol,
+      username: this.username,
+      password: (opts.showHidden || ctx.password == null) ?
+        this.password : '--------',
+      host: this.host,
+      hostname: this.hostname,
+      port: this.port,
+      pathname: this.pathname,
+      search: this.search,
+      searchParams: this.searchParams,
+      hash: this.hash
+    });
     if (opts.showHidden) {
-      ret += `  cannot-be-base: ${this[cannotBeBase]}\n`;
-      ret += `  special: ${this[special]}\n`;
+      obj.cannotBeBase = this[cannotBeBase];
+      obj.special = this[special];
+      obj[context] = this[context];
     }
-    ret += '}';
-    return ret;
+
+    return util.inspect(obj, opts).replace(/^URLTemp/,
```

Comment:
This `replace()` can be avoided by simply doing `var URL = function() {};` at the top of the function here and then simply using `new URL()` instead of `new URLTemp()`.